lib/ace/mode/csound_preprocessor_highlight_rules.js

```diff
@@ -226,70 +291,5 @@ var CsoundPreprocessorHighlightRules = function() {
 
 oop.inherits(CsoundPreprocessorHighlightRules, TextHighlightRules);
 
-(function() {
```

Comment:
why do you want to move this into the constructor? Keeping here would make it easier to refactor this to use classes in the future.

Reply:
This is so I can use the `embeddedRulePrefix` passed to the constructor in `pushRule` and `popRule`:

https://github.com/ajaxorg/ace/blob/16002cf0e16c42972ee50b1d9745ce22f91e5447/lib/ace/mode/csound_preprocessor_highlight_rules.js#L12-L17

https://github.com/ajaxorg/ace/blob/16002cf0e16c42972ee50b1d9745ce22f91e5447/lib/ace/mode/csound_preprocessor_highlight_rules.js#L45-L48

The issue I’m having is that the objects returned from `pushRule` and `popRule` don’t pick up an embedded rule prefix when they’re embedded. This is also what the previous code—

https://github.com/ajaxorg/ace/blob/36ebe1da4511e9468c6d7449bee5bb0ce4b2e276/lib/ace/mode/csound_preprocessor_highlight_rules.js#L251-L266

—was trying to solve, but that’s a very indirect way of getting an embedded rule prefix.

If there’s a way to keep this out of the constructor and also use the `embeddedRulePrefix` member variable, I’m happy to change it.

Reply:
In  pushRule and popRule you are still  using `this.embeddedRulePrefix` instead of the value from the closure. It should work the same independently from the place where these functions are defined.

Reply:
I’ve moved `pushRule` and `popRule` to where they were before.